drafts/Asynchronous-programming-made-resilient-with-MPFT.adoc

```diff
@@ -0,0 +1,251 @@
+
+But there's one in particular which is going to help you make your asynchronous methods resilient to faults: https://download.eclipse.org/microprofile/microprofile-fault-tolerance-2.1/apidocs/org/eclipse/microprofile/faulttolerance/Asynchronous.html[@Asynchronous].
+
+Let's show you what `@Asynchronous` does and how it does it.
+
+[#Asynchronous-logic]
+== What does `@Asynchronous` do?
+
+Adding the `@Asynchronous` annotation to a method does two things:
+
+1. Runs the method asynchronously
+2. Allows Fault Tolerance strategies to be applied to asynchronous methods
+
+Adding the `@Asynchronous` annotation to a method makes the method run asynchronously, meaning that rather than running straightaway on the main thread when it's called, it will run some time later, usually on another thread. To use it, you need to add the `@Asynchronous` annotation to a method, and must return either a https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletionStage.html[CompletionStage] or a https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Future.html[Future]. You cannot apply Fault Tolerance to a returned `Future` (see why <<CS-preference, here>>), so `CompletionStage` is preferred.
+
+[source,java]
+----
+@Asynchronous
+public CompletionStage<String> serviceA() { 
+
+	return CompletableFuture.completedFuture("serviceA");
+
+}
+----
+
+Note: The example uses `CompletableFuture.completedFuture()` to make the result compatible with the `CompletionStage` return type (https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html[CompletableFuture] is an implementation of the `CompletionStage` and `Future` Interfaces).
+
+What happens when the `serviceA()` method is called?
+
+Well, usually when a method is called it returns a result. However, when the `@Asynchronous` annotation is used, Fault Tolerance gets in between. When the method is called, Fault Tolerance first schedules the method to run later on a different thread, then it returns a `CompletionStage` or `Future`. This isn't the `CompletionStage` or `Future` that's returned from the method, because the method hasn't run yet, it's an object created entirely by Fault Tolerance. 
+
+image::/img/blog/FT-basic-asynchronous-execution.png[Fault Tolerance asynchronous execution flow]
+
+Sometime later, the method runs and returns a `CompletionStage` or `Future`. Now, Fault Tolerance will go and complete the `CompletionStage` or `Future` object which was returned to the caller earlier, with the value that was returned from the method. You might be wondering *_"What if the returned result is an exception? What can I do?"_*.  Read along.
```

Comment:
>Now, Fault Tolerance will go and complete the `CompletionStage` or `Future` object which was returned to the caller earlier, with the value that was returned from the method.  

To

>Now, Fault Tolerance completes the `CompletionStage` or `Future` object that was returned to the earlier, with the value that was returned from the method.

Also, add a comma after "wondering", and there's an extra space before "Read along."